rfd/0166-tpm-joining.md

```diff
@@ -0,0 +1,466 @@
+perform cryptographic operations without exposing the keys, this would provide
+a strong guarantee that the private key material cannot be exfiltrated.
+
+See https://github.com/gravitational/teleport/issues/21555 for tracking of this
+improvement.
+
+## Security Considerations
+
+### Manufacturer CA Compromise
+
+If the TPM manufacturer's CA is compromised, we can no longer trust that the
+TPM we are talking to is authentic during the enrolment process. These CAs
+are typically well protected, making this a complex attack.
+
+In addition, merely compromising the CA is not all that useful alone. The
+attacker would also need to either:
+
+- Compromise the host and intercept the TPM commands and responses.
+- Compromise the hardware supply chain and install malicious TPMs.
+
+In the case a compromise is noticed, the manufacturer can revoke the CA and
+distribute new EKCerts for effected TPMs. Users would need to update the
+configured EKCert CAs within their Teleport configuration and at this point the
+attacker would lose the ability to impersonate TPMs.
+
+### TPM Compromise
+
+If the TPM design itself is compromised, the guarantees provided by the TPM
+are lost. An attacker with access to the host would be able to extract the 
+sensitive materials from the TPM and impersonate it at will.
+
+Realistically, this is an extremely complex attack. TPMs are designed, reviewed
+and tested to reduce the risk of this happening. In addition, the attacker would
+still need to compromise the host in order to extract the sensitive materials.
+
+However, it's worth recognizing that this is not impossible. In 2023, 
+CVE-2023-1017 and CVE-2023-1018 were published, which describe a vulnerability
+in a version reference implementation code provided by the TCG for TPM vendors.
+A firmware patch was released for the affected TPMs.
+
+Using other Teleport features, such as access monitoring and IP pinning, can
+help mitigate this risk. Allowing compromise to be detected or the difficulty
+of the attack to be increased.
+
+### Certificate Exfiltration
+
+Once joined, the signed Bot certificates and private keys are stored on disk or
+in memory. Here, they can be exfiltrated by an attacker with sufficient access 
+to the host.
+
+This risk is reduced by using a short TTL for the certificates. This limits the
+amount of time an exfiltrated certificate can be used. In addition, preferring
+the in-memory storage further reduces the risk as higher privileges are
+typically required to read the memory of a process.
+
+To mitigate this entirely, we should implement "Storing Bot private key material
+in the TPM" as described in the "Future Improvements" section.
+
+### FIPS 140 Compliance
+
+See https://trustedcomputinggroup.org/wp-content/uploads/TCG_FIPS_140_Guidance_for_TPM2_0_v1r1_20170202.pdf
+for the official guidance on the compatibility of TPM 2.0 with FIPS 140.
+
+The initial implementation of TPM joining will not be FIPS 140 compatible and
```

Comment:
Given this, are we going to deny TPM joining if running in FIPS mode? It seems to me we should.

Reply:
182feff209